embdgen-core/src/embdgen/core/utils/class_factory.py

```diff
@@ -21,79 +23,86 @@ class Meta():
-    except ModuleNotFoundError: # pragma: no cover
+        from sphinx.pycode import ModuleAnalyzer  # pylint: disable=import-outside-toplevel
+    except ModuleNotFoundError:  # pragma: no cover
         return ""
     analyzer = ModuleAnalyzer.for_module(cls.__module__)
     analyzer.analyze()
     return "\n".join(analyzer.find_attr_docs().get((cls.__name__, name), [])).strip()
 
+
 CT = TypeVar("CT")
 
-def Config(name: str, doc="", optional=False):
+
+def Config(name: str, doc: str = "", optional=False) -> Callable[..., type[CT]]:
     """
     Metadata decorator for classes for the BaseFactory
     """
+
     def decorate(cls: Type[CT]) -> Type[CT]:
         local_doc = doc
-        name_prop = getattr(cls, name, None)
+        name_prop: Any | None = getattr(cls, name, None)
         if name_prop is not None and isinstance(name_prop, property):
             if name_prop.fset is None:
                 raise Exception(f"Property {name} in class {cls.__name__} has no setter defined")
-            setter_type = list(signature(name_prop.fset).parameters.values())[1].annotation
-            getter_type = signature(name_prop.fget).return_annotation
+            setter_type: Any = list(signature(name_prop.fset).parameters.values())[1].annotation
+            getter_type: Any = signature(obj=name_prop.fget).return_annotation
             if getter_type != setter_type:
                 raise Exception(f"Property {name} in class {cls.__name__} has different types for getter and setter")
             if not local_doc:
-                local_doc = name_prop.fget.__doc__ or ""
+                local_doc: str = name_prop.fget.__doc__ or ""
             hints = {name: getter_type}
         else:
-            hints = get_type_hints(cls)
+            hints: dict[str, Any] = get_type_hints(cls)
 
         if name not in hints:
             raise Exception(f"No type for member {name} in class {cls.__name__} found")
 
-        if not local_doc:
-            local_doc = try_read_doc(cls, name)
-
-        meta = Meta(name, hints[name], local_doc, optional)
+        meta = Meta(name, hints[name], local_doc or try_read_doc(cls, name), optional)
         Meta.set(cls, meta)
         return cls
+
     return decorate
 
 
 T = TypeVar("T")
+
+
 class FactoryBase(abc.ABC, Generic[T]):
-    __class_map = None
+    __class_map: Dict[Any, Type[T]] | None = None
     ALLOW_SUBCLASS = False
 
     @classmethod
     @abc.abstractmethod
-    def load(cls):
-        pass
+    def load(cls) -> None:
+        """
+        Load plugins
+        """
 
     @classmethod
-    def load_plugins(cls, root_module: ModuleType, baseclass: Type[T], key: str):
-        retval = {}
-        for (_, module_name, _) in iter_modules(list(root_module.__path__)):
-            module = import_module(f"{root_module.__name__}.{module_name}")
+    def load_plugins(cls, root_module: ModuleType, baseclass: Type[T], key: str) -> dict[Any, Any]:
+        retval: dict[Any, Any] = {}
```

Comment:
Again useless annotation. The dict annotation is derived from the assignment and the `Any` is again completely useless

Reply:
Annotations are in general useless and only making everything slower. 😆 I would use Go for this project, for example.

Reply:
See above.